src/Microsoft.AspNet.OData/Results/ResultHelpers.cs

```diff
@@ -60,7 +60,8 @@ public static void AddEntityId(HttpResponseMessage response, Func<Uri> entityId)
         {
             if (response.StatusCode == HttpStatusCode.NoContent)
             {
-                response.Headers.TryAddWithoutValidation(EntityIdHeaderName, entityId().ToString());
+                Uri location = entityId();
```

Comment:
why use an intermediate variable can resolve this problem?

Reply:
@xuzhg The intermediate problem doesn't resolve the problem. If you look at the line that follows, we proceed to call `location.AbsoluteUri` and that works, as opposed to `entityId().ToString()`. I could however ditch the intermediate variable and use `entityId().AbsoluteUri` in the line that follows if this is causing confusion

Reply:
@gathogojr Your first sentence is a bit confusing but yes, I would also prefer to use `entityId().AbsoluteUri` instead of introducing another variable.